src/core/dep-graph.ts

```diff
@@ -327,39 +308,60 @@ class DepGraphImpl implements types.DepGraphInternal {
     return node;
   }
 
-  private pathsFromNodeToRoot(nodeId: string): types.PkgInfo[][] {
+  private pathsFromNodeToRoot(
```

Comment:
I don't follow why this change was needed? How can we have cycles when traversing from a node to the root?

Reply:
if this cycle exists:
`A -> B -> C -> B`
(and B is exactly the same node)

than B have 2 parents: A and C, so more than over traversal up into C is a cyclic one.

we can (and should) also see if the current tests catch this. @aviadhahami can you please try? don't do cycle check in `pathsFromNodeToRoot` and see what happens

Reply:
ah.. I think I get it. 😢

Reply:
Will add as a test too

Reply:
already tested by this, no? https://snyk.slack.com/archives/CFE1B1M8B/p1588757577066900?thread_ts=1588752090.061900&cid=CFE1B1M8B

just need to check that the test fails without the code change. (I hope I did that when writing the test, but it was few months ago)

Reply:
Will try this soon 👌
Not next to my laptop atm 💈

Reply:
@michael-go link is wrong imo?

Reply:
oops sorry! fixed now

Reply:
Tested - tests throw cyclic deps warning :tada: 
you can checkout the code here [`tests/failin-cyclic-graphs`](https://github.com/snyk/dep-graph/tree/tests/failin-cyclic-graphs) and run `npm t` or see the SS:

![image](https://user-images.githubusercontent.com/7353632/81193660-57f11200-8fc4-11ea-8319-919d81dbf8a2.png)
![image](https://user-images.githubusercontent.com/7353632/81193672-5de6f300-8fc4-11ea-928b-87ad36e052ea.png)

Reply:
@aviadhahami this is not exactly what I meant. remove the `hasCycles()` check (https://github.com/snyk/dep-graph/blob/tests/failin-cyclic-graphs/src/core/dep-graph.ts#L134) and then run it - and the error we expect to see is some sort of call-stack overflow due to cyclic function calls

Reply:
oh, ok :+1:  omw

Reply:
yup, it throws:
```text
RangeError: Maximum call stack size exceeded
Exception in PromiseRejectCallback:
/home/aviad/Documents/code/snyk/dep-graph/src/legacy/index.ts:207
        const subtree = await buildSubtree(depGraph, depInstId, eventLoopSpinner, maybeDeduplicationSet);

```